planner/core/stats.go

```diff
@@ -217,6 +218,20 @@ func init() {
 	cardinality.GetTblInfoForUsedStatsByPhysicalID = getTblInfoForUsedStatsByPhysicalID
 }
 
+// findTableByTblOrPartID looks for table.Table for the given id in the InfoSchema.
+// The id can be either a table id or a partition id.
+// If the id is a table id, the corresponding table.Table will be returned, and the second return value is nil.
+// If the id is a partition id, the corresponding table.Table and PartitionDefinition will be returned.
+// If the id is not found in the InfoSchema, nil will be returned for both return values.
+func findTableByTblOrPartID(is infoschema.InfoSchema, id int64) (table.Table, *model.PartitionDefinition) {
```

Comment:
Can we put in into the member function of InfoSchema? it can help us to reuse it next time.?

Reply:
The `InfoSchema` has 3 implementations. Making this a member method will make it 3 repeated functions.
So I just moved this function into the `infoschema` package.